src/value.cc

```diff
@@ -1241,7 +1241,7 @@ void FreeGIArgument(GITypeInfo *type_info, GIArgument *arg, GITransfer transfer,
 
     case GI_TYPE_TAG_ERROR:
     {
-        g_error_free((GError *)arg->v_pointer);
+        // no g_error_free as the ptr is taken directly from box
```

Comment:
This logic is a bit far in my memory, could you point me out where we free this memory now (or why it's not needed to free it)?

Reply:
When receiving a return value from C, `GIArgumentToV8` is used to convert it to a JavaScript value. Changing the various `mustCopy` to `ownership` allows the C value to not be copied (i.e. skipping `g_boxed_copy`) but `owns_memory` set so it's freed when GC'ed (`g_boxed_free`).

For you commented part, `FreeGIArgument` is called when function calls return to free converted C values, after `V8ToGIArgument` is used to convert JavaScript values to C values. `GError`, similar to unions and structs, are converted by simply retrieving the pointer from JavaScript box via `PointerFromWrapper`. So the resulting pointer should not be freed with `g_error_free`.